src/Input/Silk.NET.Input.Glfw/ReadOnlyListEnumerator.cs

```diff
@@ -17,28 +17,24 @@ public ReadOnlyListEnumerator(IReadOnlyList<T> list)
         {
             _list = list;
             _i = 0;
-            Current = default!;
         }
         public bool MoveNext()
         {
             if (_i >= _list.Count)
             {
-                Current = default!;
                 return false;
             }
 
-            Current = _list[_i];
             _i++;
             return true;
         }
 
         public void Reset()
         {
             _i = 0;
-            Current = default!;
         }
 
-        public T Current { get; private set; }
+        public T Current => _i >= _list.Count || _i < 0 ? default! : _list[_i];
```

Comment:
I think those checks are just a waste, calling `Current` can even throw stack overflow if MoveNext was false

Reply:
I don't see the stack overflow in this code?

Reply:
that was just an example, apologies, I was meaning to say it doesn't matter what happens when `_i` is out of bounds -> MoveNext has returned false
So the checks are not needed

Reply:
```suggestion
        public T Current => _list[_i];
```